### PR TITLE
Feature/api for plugins per channel

### PR DESCRIPTION
**Pull request by korycins**
I want to merge this change because...

# Impact

* [ ] New migrations
* [x] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Review by fowczarek — APPROVED** [review]

File: saleor/graphql/plugins/types.py [hunk 1]
```diff
@@ -45,34 +22,61 @@ class Meta:
...
+    )
+    name = graphene.String(description="Name of the plugin.")
+    description = graphene.String(description="Description of the plugin.")
+    global_configuration = graphene.Field(
+        PluginConfiguration,
+        description="Configuration for plugins which configuration is not channel-specific.",
+    )
+    channel_configurations = graphene.List(
+        graphene.NonNull(PluginConfiguration),
+        description="Configuration of plugins for each existing channel.",
```
[the remark this answers is not in the document]
Comment (maarcingebala) [reply saleor/graphql/plugins/types.py:53]: Also this field will never return null I think, but always a list (may be an empty list).

[the remark this answers is not in the document]
Comment (korycins) [reply saleor/graphql/plugins/types.py:53]: @maarcingebala  Not sure if understand.
Each plugin has two ways of configuration: per channel and global(which is not related to channel).  Assuming that you have a plugin with the global configuration, the channel_configurations will be None. 
Is it something that you don't like or thinking that this is not the best way?

File: saleor/graphql/plugins/types.py [hunk 2]
```diff
@@ -45,34 +22,61 @@ class Meta:
...
+
+class Plugin(graphene.ObjectType):
+    id = graphene.Field(
+        type=graphene.String, required=True, description="Identifier of the plugin."
+    )
+    name = graphene.String(description="Name of the plugin.")
+    description = graphene.String(description="Description of the plugin.")
+    global_configuration = graphene.Field(
+        PluginConfiguration,
+        description="Configuration for plugins which configuration is not channel-specific.",
```
[the remark this answers is not in the document]
Comment (korycins) [reply saleor/graphql/plugins/types.py:49]: Yeap, as described in previous comment. Each plugin can have globalConfiguration OR channelConfigurations.

File: saleor/graphql/plugins/types.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (maarcingebala) [reply saleor/graphql/plugins/types.py:53]: I think it's better to return an empty list in that case, but we can ask @jwm0 if it makes any difference.

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**